src/service/cache/value-cache.service.js

```diff
@@ -0,0 +1,447 @@
+const fs = require('node:fs/promises')
+const path = require('node:path')
+
+const { nanoid } = require('nanoid')
+
+const { createFolder } = require('../utils')
+const DeferredPromise = require('../deferred-promise')
+
+const BUFFER_MAX = 250
+const BUFFER_TIMEOUT = 300
+const RESEND_IMMEDIATELY_TIMEOUT = 100
+
+const VALUE_FOLDER = 'values'
+const ERROR_FOLDER = 'values-errors'
+
+const BUFFER_FILE_NAME = 'buffer.tmp'
+
+/**
+ * Local cache implementation to group events and store them when the communication with the North is down.
+ */
```

Comment:
we should add a readme.md explaining the design of the cache and refer to it. Please make an issue for this